setup/util/reload_posix.go

```diff
@@ -0,0 +1,45 @@
+// +build linux freebsd darwin
```

Comment:
If we have a way of sharing code with the main binary, we could avoid copying this (not sure if thats straightforward?)

Reply:
We can share code--the problem is that code calls `os.Exit` and logs some stuff that doesn't really make sense in this context.

Looking at this again, though, I can refactor to share code here without too much trouble. We do also have similar around `discoverLogLocation`, and that might be tougher, since that's more integrated into other collector processing. (Ideally, though, that functionality will migrate into the setup binary entirely over time, I think?)